### PR TITLE
Allow to use unix socket for redis backend

### PR DESCRIPTION
**Pull request by deees**

### CODE REVIEW

File: src/Backend/Factory.php [hunk 1]
```diff
@@ -56,7 +56,11 @@ public function buildRedisCache($options)
         }
 
         $redis = new \Redis();
-        $redis->connect($options['host'], $options['port'], $timeout);
+        if (empty($options['unix_socket'])) {
+            $redis->pconnect($options['host'], $options['port'], $timeout);
```
Comment (tsteur) [inline src/Backend/Factory.php]: Is the change to pconnect really needed? As far as I can see this would be a persistent connection

Comment (deees) [inline src/Backend/Factory.php]: Sorry, not related to this pull request, will remove.

https://github.com/piwik/plugin-QueuedTracking/pull/47#discussion_r87799311

**Comment by tsteur** [comment]
This looks good to merge to me. I will prepare a PR for Piwik 3 and add the new config setting to the template so users will know that they can use it.

**Comment by tsteur** [comment]
Thanks for the PR 👍

**Pull request merged**